ha/core/event_analyzer/event_analyzerd.py

```diff
@@ -86,6 +88,8 @@ def _initalize_watcher(self, system_health: SystemHealth) -> dict:
                 event_parser = event_parser_instance,
                 subscriber = system_health
             )
+            self._iem_watcher = watcher_list[watcher]
```

Comment:
we should do this assignment only if [watcher] == iem
Otherwise add a TODO mentioning the same; for when multiple watchers will be there .